src/transformer/url_basename.cpp

```diff
@@ -0,0 +1,40 @@
+// Unless explicitly stated otherwise all files in this repository are
+// dual-licensed under the Apache-2.0 License or BSD-3-Clause License.
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+#include "transformer/url_basename.hpp"
+
+namespace ddwaf::transformer {
+
+bool url_basename::transform_impl(cow_string &str)
+{
+    size_t path_end = 0;
```

Comment:
I find this name a bit odd. It will become the path end but most of the time it isn't. It is just an iterator